modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/typescript/index.ftl

```diff
@@ -50,66 +33,50 @@ function endsWith(str: string, match: string): boolean {
 	);
 }
 
-const nullableSuffix = ' | null';
-const optionalSuffix = ' | undefined';
+function startsWith(str: string, match: string): boolean {
+	return str.substring(0, match.length) === match;
+}
+
 const arrayPrefix = 'Array<';
 const arraySuffix = '>';
 const mapPrefix = '{ [key: string]: ';
 const mapSuffix = '; }';
+const nullableSuffix = ' | null';
+const optionalSuffix = ' | undefined';
+const primitives = new Set([
+	'string',
+	'boolean',
+	'double',
+	'integer',
+	'long',
+	'float',
+	'number',
+	'any',
+]);
+const typeMap: {[index: string]: any} = {
+<#list schemas?keys?sort as schema>
+	${schema},
+</#list>
+};
 
 export class ObjectSerializer {
```

Comment:
Move this class to a separate file and all the other private methods so that this index file only does exports and is much cleaner. Also in a separate PR.